crates/cxx-qt-gen/src/naming/cpp.rs

```diff
@@ -332,6 +332,64 @@ mod tests {
         ];
     }
 
+    #[test]
+    fn test_syn_type_invalid() {
+        let ty = parse_quote! { (A) };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert!(syn_type_to_cpp_type(&ty, &type_names).is_err());
+
+        let ty = parse_quote! { Option<A> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert!(syn_type_to_cpp_type(&ty, &type_names).is_err());
+
+        let ty = parse_quote! { Result<A> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert!(syn_type_to_cpp_type(&ty, &type_names).is_err());
+
+        let ty = parse_quote! { Pin<> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert!(syn_type_to_cpp_type(&ty, &type_names).is_err());
+    }
+
+    #[test]
+    fn test_syn_type_to_cpp_type_templates() {
+        let ty = parse_quote! { SharedPtr<A> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert_eq!(
+            syn_type_to_cpp_type(&ty, &type_names).unwrap(),
+            "::std::shared_ptr<A1>"
+        );
+
+        let ty = parse_quote! { WeakPtr<A> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert_eq!(
+            syn_type_to_cpp_type(&ty, &type_names).unwrap(),
+            "::std::weak_ptr<A1>"
+        );
+
+        let ty = parse_quote! { CxxVector<A> };
+        let mut type_names = TypeNames::default();
+        type_names.mock_insert("A", None, Some("A1"), None);
+        assert_eq!(
+            syn_type_to_cpp_type(&ty, &type_names).unwrap(),
+            "::std::vector<A1>"
+        );
+    }
```

Comment:
i think these ones can go into the other test `test_syn_type_to_cpp_type` that has a helper so these would be
```
{ SharedPtr<T } => "::std::shared_ptr<T>",
```

Or did you specifically need the A -> A1 conversion also to happen? If so you could add
`type_names.mock_insert("A", None, Some("A1"), None);`  to the `test_syn_types_to_cpp_types` then use `SharedPtr<A>` etc

Reply:
Used them in the macro helper instead now